src/Pos/Explorer/Web/Server.hs

```diff
@@ -407,29 +435,32 @@ topsortTxsOrFail f =
     maybeThrow (Internal "Dependency loop in txs set") .
     topsortTxs f
 
+-- | Convert server address to client address,
+-- with the possible exception (effect).
 cAddrToAddr :: MonadThrow m => CAddress -> m Address
-cAddrToAddr cAddr =
-    fromCAddress cAddr &
-    either (const $ throwM $ Internal "Invalid address!") pure
+cAddrToAddr cAddr = either exception pure (fromCAddress cAddr)
+  where
+    exception = const $ throwM $ Internal "Invalid address!"
 
+-- | Convert server transaction to client transaction,
+-- with the possible exception (effect).
 cTxIdToTxId :: MonadThrow m => CTxId -> m TxId
-cTxIdToTxId cTxId =
-    fromCTxId cTxId &
-    either (const $ throwM $ Internal "Invalid transaction id!") pure
+cTxIdToTxId cTxId = either exception pure (fromCTxId cTxId)
+  where
+    exception = const $ throwM $ Internal "Invalid transaction id!"
 
 getMainBlock :: ExplorerMode m => HeaderHash -> m (MainBlock SscGodTossing)
 getMainBlock h =
     DB.getBlock h >>=
     maybeThrow (Internal "No block found") >>=
     either (const $ throwM $ Internal "Block is genesis block") pure
-{-}
-getTxExtra :: ExplorerMode m => TxId -> m (Maybe TxExtra)
-getTxExtra id =
-    MM.lookupM EX.getTxExtra id =<< getTxExtra
--}
-getTxExtraOrFail :: ExplorerMode m => TxId -> m TxExtra
-getTxExtraOrFail id =
-    maybeThrow (Internal "Transaction not found") =<< getTxExtra id
+
+-- | Get transaction extra from the database, and if you don't find it
+-- throw an exception.
+getTxExtraOrFail :: MonadDB m => TxId -> m TxExtra
+getTxExtraOrFail id = getTxExtra id >>= maybeThrow exception
```

Comment:
Use more descriptive name then `id`. Id is usually identify function and looks confusing at first. `txId` would do